### DISCUSSION TITLE
General (kinda off-topic), but I hope you guys would be willing to help me...

### DISCUSSION THREAD

**Discussion by poorandunlucky** [gabonator/LA104#60] [General]
So I'm an electronics hobbyist, and I'm looking for an oscilloscope...

Right now I have the $30 8 digital + 1/2 analog channels logic analyzer that's very noisy, and the connectors are very fiddly, so I rarely use it, also needs sigrok/pulseview...

I'm kind of on the market for a DSO/MSO, and the LA104 seems like it might be a good option because it claims to be a 100 MHz oscilloscope with a logic analyzer...  For the price, $100, it seems like it could be a good deal, but I'm not sure if it's one or not...

I could get a real oscilloscope, Hantek, or Siglent, 100 MHz, 2 channels, maybe with a waveform generator, maybe with a touchscreen, for $300...  I could get a real MSO, but then that's $700, plus the logic probes, so around CA$850/900...

It's a good investment because it probably won't lose a lot of resale value, but I'm a hobbyist, so the use I'm going to get out of it...  I don't know.  I wish I had the cash, it might spark my interest even more, but I don't really...  I'd rather get a new phone than that at this point...

So I'm just wondering if it's worth $100 to use as an oscilliscope, and logic analyzer...  Something I can keep on my desk to see voltage across time, and decode logic signals when playing with my MCUs (I'm an ESP person) (with the NodeMCU Lua firmware).

You guys seem like pretty equipped already, and like it's a toy, kinda like some engineers (say Ben Heck) likes to play Doom on his TI-83 calculator...

I'm still havinga a hard time with transistors...  So a scope/LA/AWG in one machine is something I'm often looking at, but it's really expensive for me (I'm just a kid living on my own, too)...  You seem experienced with this machine, you aren't looking for my YouTube view, or my affiliate link purchase, you've torn the thing to pieces, and put it back together, and got years of experience in electronics...

I'm just going on a limb here, and asking for your help here because I want an honest opinion, and Github is kinda off the beaten path, I don't want to talk to some jerks on a forum, I've had enough of that, and you guys seem OK...

Would it be a good purchase for me, or is it just going to frustrate me?  I can try it from Amazon, no problem, I got Amazon Prime and all, but for a bit less than $200 more, I can get a real, 100-150 MHz scope, but from AliExpress...  I don't have space on my desk, and it's not a LeCroy, or Keysight, so I'm not sure how big a hard-on I'd have from it, either...  Which brings me back to the LA104...

Besides being a cool toy for someone who's got a scope, and knows what they're doing, is it a good spend of $100 for a scope/LA, and I think it outputs signals, too...  Like would I be able to characterize a transistor with it?  Or make a RC oscilator and see the harmonics?

Is it a good scope/LA/AWG, or is it a cool toy with annoying buttons for someone who got a desk scope, and could use a cheap portable device?

I thank you for reading all this, and I thank you in advance is you try to help me light-up the paths before me...  Thanks.

**Comment by gabonator** [comment]
LA104 is just logic analyser, does not have any analog inputs. Even if the LA104 is very clever device and can be used for ton of various tasks, I have never used it as logic analyser. The interface was too complex for me and I found it quite difficult to use it even for simple analysis. In general there are not many situations where you would need to use logic analyser - Most buses have 2 or 3 wires and can be easily debugged with oscilloscope. Oscilloscope shows you voltage levels instead of just ones/zeroes shown by analyser. Precise voltage level is very important - when some sensor is not working properly, it can be caused by noise, or by wrong voltage levels or something else.
I remember when I was in the same situation as you are 14 years ago - looking for cheap and decent oscilloscope. At that time I already had some experience with oscilloscopes - as a student at technical university we had to learn how they work and draw a simplified diagram of them. At that time we were working with some Tektronix oscilloscopes with CRT display which took half of the desk. Modern LCD oscilloscopes were incredibly expensive and far beyond my budget. But there were some Chinese companies that made quite decent cheap and small oscilloscopes. So my first oscilloscope was Owon pds5022 - it had just 25MHz bandwidth, but it was still perfectly sufficient for repairing amplifiers, reverse engineering serial protocols, etc. If I had to buy some osc nowadays, I would look for brands like Rigol. Higher the sampling rate, higher the price, so for me anything with more 20MHz is fine.... Actually I do not own any professional oscilloscope, my Owon is dead (display broke and the encoder knobs did not work correctly after few years) and I use only DS203, mostly as usb oscilloscope with my laptop. Having a battery powered oscilloscope makes a lot of things much easier, you do not need to think about ground loops, you can use it anywhere - e.g. sniffing the canbus of your car.... DS203 (made by the same company as LA104) is not produced anymore, because the FPGA chip they used is not manufactured anymore, and the successor DS213 has some serious bugs in the FPGA code and was not fixed yet. 
So my suggestion is:
- do not buy those $20 oscilloscopes from ebay - it is just MCU board with display without proper analog frontend where they used an analog input of the MCU for sampling 
- choose lower sample rates to save money
- battery powered
- avoid USB only oscilloscopes
- 2 channels is enough
- always read reviews (e.g. on eevblog.com)
- don't hesitate to choose lowend chinese oscilloscopes (but with with proper analog fontend and FPGA based sampler), again to save money

**Comment by poorandunlucky** [comment]
Yeah, yesterday I stumbled on the Rigol MSO5000 series, they’re the price of a laptop with the options, but they seem to have everything,,,,  Scope, LA, spectrum analyzer, waveform generator, …  They start relatively affordable, and I live in a big city, so it would be an easy sell let’s say I needed the money for some reason, …

I think it’s best to wait until I actually need one, there’s a small 10 V analog channel on my LHT00SU1, 8 MHz, very noisy, but for the stage I’m in, I think it’s good enough…  I would love the MSo5000 and would buy right now if I could afford it, but I can’t, plus I’m not so good at electronics, I’m doing a lot of things at the same time, 3D printing, CAD, skateboarding, cooking, etc, but I think I’ll keep dealing with it for now, maybe fix the connectors, make a RF shield, but Sigrok/Pulseview are good enough, and it’s helped me already trying to bitbang a NRF24L01+…

I just don’t think the other scopes are worth it for the price, so I’m glad I found the MSO5000, it’s at least given me a target in mind, I can stop looking, and start doing…

Also after watching your videos on your firmware, I realized there weren’t any analog channels, I figured maybe it was doing digital acquisition of voltage points or something, but apparently it’s just high/low, but you’ve turned it into a nice little diagnostic tool from your video…  I’m sure it’s very useful to quite a few people, and just like the jyers firmware for the Ender 3 V2 (Marlin is made for a LCD matrix display, and most of the function were unavailable on the DWIN display on the Ender V2, but he’s managed to turn the firmware that was essentially unusable on the V2 because of the screen, and made all functions available.  He’s just done real good work, your involvement reminds me of his, you just took something kinda crappy, and made it really useful, and amazing, and I find that inspiring.  😊

Anyway, thanks for the reply, and mentioning Rigol that I was kinda on the fence about before yesterday, but you mentioning it really sets my sights, and I know where to look if/when the time comes I get a real sope.

Thanks,  😊

P.S.: And thanks for the warning about the DS213!
